sw/device/silicon_creator/lib/crypto/ecdsa_p256/meson.build

```diff
@@ -11,3 +12,96 @@ sw_lib_crypto_otbn_sources += {
     'p256.s',
   ),
 }
+
+# Setup for building OTBN assembly. OTBN software is build with a separate
+# toolchain, which is called as an external target from Meson in
+# subdirectories.
+
+prog_otbn_build = meson.source_root() / 'util/otbn_build.py'
+
+otbn_build_command = [
+  prog_env,
+  'OTBN_AS=@0@'.format(prog_otbn_as),
+  'OTBN_LD=@0@'.format(prog_otbn_ld),
+  'RV32_TOOL_OBJCOPY=@0@'.format(prog_objcopy.path()),
+  'RV32_TOOL_AS=@0@'.format(prog_as.path()),
+  'RV32_TOOL_LD=@0@'.format(prog_ld.path()),
+  prog_python,
+  prog_otbn_build,
+  '--out-dir',
+  '@OUTDIR@',
+  '@INPUT@',
+]
+otbn_build_depend_files = [
+  prog_otbn_as,
+  prog_otbn_ld,
+  prog_objcopy.path(),
+  prog_as.path(),
+  prog_ld.path(),
+  prog_otbn_build,
+]
+
+# OTBN BUILD procedure
+sw_lib_crypto_otbn = {}
+foreach sw_lib_crypto_otbn__app_name, sw_lib_crypto_otbn__app_sources : sw_lib_crypto_otbn_sources
```

Comment:
nit: The loop variables could be simplified to `app_name` and `app_sources`.  I actually don't think the prefixing of `sw_lib_crypto_otbn__` makes this block easier to read.

Reply:
I adjusted the local variables inside the loop, too, and removed the `sw_lib_crypto_otbn__` prefix from e.g. `sw_lib_crypto_otbn__app_output_files` -- is that compatible with Meson's local-variables rules, or do they persist outside the loop?